text/0000-decouple-dependencies.md

```diff
@@ -0,0 +1,403 @@
+  present in `buildpack.toml`.
+- We will add a dependency version validation section to `buildpack.toml`
+  metadata, this can be used to state that a buildpack version only supports
+  certain ranges of a given tool, such as Java `11.*` or Node.js `16.*`.
+- Metadata can be provided by anyone. A user can add custom metadata, or source
+  from a third party project. Paketo will provide an official set of metadata
+  against which we will test the Paketo buildpacks. It will be distributed via
+  images in an image registry (Docker Hub).
+- Buildpacks do not care how dependency metadata is distributed, that is a
+  separate concern, instead, they just read metadata from a specified location.
+  It is on the platform to mount the metadata that it fetched at that location.
+- The actual dependencies are accessed via the metadata and that can happen
+  over any protocol (HTTPS/SFTP/FILE) and be distributed in any format
+  (archive/image).
+- Dependency metadata will be removed from `buildpack.toml`
+
+
+## Metadata Format
+The metadata presented to a buildpack will be structured as a directory of
+dependencies.
+
+Like it does now, each dependency will have an id but unlike the present
+situation, the id is namespaced. An id is composed of an organization name and
+a dependency name. It follows the [reverse domain name notation](https://en.wikipedia.org/wiki/Reverse_domain_name_notation) and the
+dependency name is defined as the final item, so `com.example.dep-a` would have
+an organization of `com.example` and a dependency of `dep-a`. It is case
+insensitive, so `dep-a` is no different than `Dep-A`. This is to reduce the
+possibility of [typosquatting](https://en.wikipedia.org/wiki/Typosquatting).
+Allowed characters are [the same as for a valid hostname](https://en.wikipedia.org/wiki/Hostname#Syntax).
+
+The directory structure will contain a folder for each dot-separated segment of
+the dependency’s organization name and in the lowest level directory there will
+be a file named after the dependency name with the extension `toml`, this is
+because the metadata file will be TOML format. For example, with
+`com.example.dep-a`, there would be the following folder structure:
+
+```
+com
+└── example
+    └── dep-a.toml
+```
+
+As mentioned previously, each individual metadata file has a file name
+consisting of the dependency name with an extension of `toml`. The internal
+format consists of a single table called `versions` which is an array of tables
+containing all of the versions for that particular dependency. Each version
+entry requires a `uri`, `version`, `checksum`, `arch`, `os`, and `license`. It
+may also have `name`, `purl`, `strip-components`, and `cpes` although these are
+optional. The `cpes` entry is an array of strings identifying all of the CPEs
+for that dependency. The `license` is itself an array of tables containing
+`type` and `uri` of the license for the dependency.
```

Comment:
Are these the same mandatory and/or optional values they currently are for the `buildpack.toml`? If they differ, like this RFC introducing additional ones like `arch` and `os` or if there are fields that changed from mandatory to optional or vice versa this would be worth pointing out explicitly.

Reply:
Agree that it's worth being explicitly. FWIW I was assuming this RFC wouldn't change the syntax/semantics of the existing metadata - it just allows it to be located at a new location on the filesystem.

Reply:
Yeah I don't think that any default values with change. I think that introduction of things like `arch` and `os` are future proofing us for the removal of stacks and introduction of ARM capable buildpacks.